cfme/containers/provider/__init__.py

```diff
@@ -91,15 +91,41 @@ class hawkular(Tab, BeforeFillMixin):  # NOQA
             Version.lowest(): Input('hawkular_api_port'),
             '5.9': Input('metrics_api_port')
         })
+
+        validate = Button('Validate')
+
+    @View.nested
+    class prometheus(Tab, BeforeFillMixin):  # NOQA
+        TAB_NAME = VersionPick({
+            '5.9': 'Metrics'})
+        sec_protocol = VersionPick({
+            '5.9': BootstrapSelect(id='metrics_security_protocol')})
+        trusted_ca_certificates = VersionPick({
+            '5.9': TextInput('metrics_tls_ca_certs')})
+        hostname = VersionPick({
+            '5.9': Input('metrics_hostname')})
+        api_port = VersionPick({
+            '5.9': Input('metrics_api_port')})
```

Comment:
@shalomnaim1 I still do not believe this is the correct approach to modeling these endpoint tabs.  I pulled your ocp-v3 provider yaml, and this branch, in order to demonstrate what this creates in the views.

This is a completely duplicate view of the hawkular endpoint, which should just be converted to a metrics tab, where the name is version picked for 'Hawkular' in 5.8 and 'Metrics' in 5.9.

```python
In []: ocp = providers.get_crud('ocp-v3')

In []: app = store.current_appliance

In []: ocp.endpoints.keys()
Out[]: ['default', 'prometheus', 'alerts']
```

Above isn't correct in my view, the openshift endpoints should be `default`, `metrics`, and `alerts` for providers in 5.9, and `default` and `metrics` in 5.8. Prometheus is simply a type for the metrics endpoint. This is ultimately driven by the yaml, and it matching the endpoint class name.

The changing tab name needing to match the endpoint name in the yaml can be solved with a version picked name for the common metrics endpoint class that would replace the hawkular endpoints class 

```python
view = navigate_to(ocp, 'Add')
prom_view = getattr(ocp.endpoints_form(parent=view), 'prometheus')

prom_view.__class__()
Out[]: prometheus()

prom_view.cls_widget_names()
Out[]: ('sec_protocol', 'trusted_ca_certificates', 'hostname', 'api_port', 'validate')
```
Above is how endpoints are filled in BaseProvider.create(), through the endpoint_form attribute and the endpoint key from the yaml.

Note the widget names for the prometheus view tab, for ocp-v3 with 'prometheus' in yaml

Look below at how ocp-v2, with 'hawkular' in yaml has exactly the same widgets, but is a different class

```
ocp2 = providers.get_crud('ocp-v2')
ocp2.endpoints.keys()
Out[]: ['default', 'hawkular']

hawk_view.__class__()
Out[]: hawkular()

hawk_view = getattr(ocp.endpoints_form(parent=view), 'hawkular')
hawk_view.cls_widget_names()
Out[]: ('sec_protocol', 'trusted_ca_certificates', 'hostname', 'api_port', 'validate')
```

This demonstrates the duplicate classes that you've created here in this PR.

There may be some small complications from the class simplification when it comes to having to version pick some things, but I don't believe that a simplified class structure for a common 'metrics' endpoint is complex or unreasonable to require at such a core level of the framework.

Reply:
Take a look at this now please, I updated the yaml as well so all OCP providers have metrics endpoint instead of Hawkular/Prometheus